zerver/lib/markdown/__init__.py

```diff
@@ -1845,32 +1874,42 @@ def handleMatch(self, m: Match[str]) -> Optional[Element]:
             else:
                 # Don't highlight @-mentions that don't refer to a valid user
                 # group.
-                return None
+                return None, None, None
 
             el = Element("span")
             el.set("class", "user-group-mention")
             el.set("data-user-group-id", user_group_id)
             text = f"@{name}"
             el.text = markdown.util.AtomicString(text)
-            return el
-        return None
+            return el, match_span[0], match_span[1]
+        return None, None, None
 
 
-class StreamPattern(CompiledPattern):
+class StreamPattern(markdown.inlinepatterns.InlineProcessor):
+    def __init__(self, compiled_re: Pattern, md: markdown.Markdown) -> None:
+        # This is similar to the superclass's small __init__ function,
+        # but we skip the compilation step and let the caller give us
+        # a compiled regex.
+        self.compiled_re = compiled_re
```

Comment:
That shouldn’t be necessary. `re.compile` already accepts a compiled regex and skips the compilation step.

Reply:
superclass constructor calls `re.compile` with `DOTALL` and `UNICODE` flag which fail if the pattern is already compiled so I had to define the constructor in order to accept pre-compiled pattern.